### PR TITLE
Enable tracing for DeviceMesh

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #460
* #476
* #459
* __->__ #456

### CODE REVIEW

**Comment by mrshenli** [comment]
I am going to remove eager-mode support from CommTensor. We can always add it back when necessary. At this moment, CommTensor's eager-mode support is not helping anything.

**Comment by mrshenli** [comment]
land after https://github.com/pytorch/pytorch/pull/84963

File: spmd/tensor/device_mesh.py [hunk 1]
```diff
@@ -266,12 +267,21 @@ def scatter(
         Returns:
             A :class:`torch.Tensor` object
         """
-        to_scatter = [tensor.contiguous() for tensor in scatter_list]
+        # CommTensor does not change earger mode behavior. During tracing, it
```
Comment (wanchaol) [inline spmd/tensor/device_mesh.py:270]: nit: eager

File: spmd/tensor/device_mesh.py [hunk 2]
```diff
@@ -292,12 +302,20 @@ def broadcast(self, tensor: torch.Tensor, mesh_dim: int = 0):
         if dim_group is not GroupMember.WORLD:
             src_for_dim = get_global_rank(dim_group, 0)
 
-        return broadcast(tensor.contiguous(), src=src_for_dim, group=dim_group)
+        # CommTensor does not change earger mode behavior. During tracing, it
```
Comment (wanchaol) [inline spmd/tensor/device_mesh.py:305]: ditto

**Review by wanchaol — APPROVED** [review]
lgtm

**Comment by wanchaol** [comment]
it seems if using ghstack, you need to use `ghstack land`

**Comment by mrshenli** [comment]
thanks for the info. I will create a new PR for this.

**Pull request merged**